diff --git a/features.module b/features.module
index ae88b67..6d731ec 100644
--- a/features.module
+++ b/features.module
@@ -278,7 +278,13 @@ function features_flush_caches() {
       features_get_modules(NULL, TRUE);
     }
   }
-  return array('cache_features');
+
+  // The update to add the table needs to have run.
+  include_once DRUPAL_ROOT . '/includes/install.inc';
+  if (drupal_get_installed_schema_version('features') >= 7200) {
+    return array('cache_features');
+  }
+  return array();
 }
 
 /**
