diff --git a/print_ui/print_ui.module b/print_ui/print_ui.module
index 3affdb3..0c4f9c7 100644
--- a/print_ui/print_ui.module
+++ b/print_ui/print_ui.module
@@ -623,8 +623,7 @@ function print_ui_insert_link($link, $args = array()) {
 function print_ui_link_allowed($link, $args) {
   if (isset($args['view_mode'])) {
     $view_mode = $args['view_mode'];
-    if ((($view_mode == 'teaser') && !variable_get('print_' . $link['format'] . '_link_teaser', PRINT_UI_LINK_TEASER_DEFAULT))
-      || !in_array($view_mode, array('full', 'teaser'))) {
+    if ((($view_mode == 'teaser') && !variable_get('print_' . $link['format'] . '_link_teaser', PRINT_UI_LINK_TEASER_DEFAULT))) {
       // If the teaser link is disabled
       return FALSE;
     }
