From f92a0facc1586fc1d7cfc08f1ff4be463c9f0045 Mon Sep 17 00:00:00 2001
From: Dane Powell <git@danepowell.com>
Date: Mon, 13 Apr 2015 15:25:06 -0500
Subject: [PATCH] Issue #2148453 by Dane Powell: Fixed image styles not
 reverting.

---
 includes/features.image.inc | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/includes/features.image.inc b/includes/features.image.inc
index b2058b7..bf89191 100644
--- a/includes/features.image.inc
+++ b/includes/features.image.inc
@@ -73,11 +73,17 @@ function image_features_export_render($module_name, $data, $export = NULL) {
  */
 function image_features_revert($module) {
   if ($default_styles = features_get_default('image', $module)) {
-    foreach (array_keys($default_styles) as $default_style) {
-      if ($style = image_style_load($default_style)) {
+    foreach ($default_styles as $default_style_name => $default_style) {
+      if ($style = image_style_load($default_style_name)) {
         if ($style['storage'] != IMAGE_STORAGE_DEFAULT) {
           image_default_style_revert($style);
         }
+        else {
+          // Verify that the loaded style still matches what's in code.
+          if ($default_style['effects'] !== $style['effects']) {
+            image_default_style_revert($style);
+          }
+        }
       }
     }
   }
--
2.1.0
