diff --git a/ife.theme.inc b/ife.theme.inc
index 59394e0..94b4316 100644
--- a/ife.theme.inc
+++ b/ife.theme.inc
@@ -52,6 +52,18 @@ function theme_ife_form_element($variables) {
   if (isset($variables['element']['#id'])) {
     $error = ife_errors('get', $variables['element']['#id']);

+    // If the error message hasn't been set, it may be available from another module.
+    if (empty($error)) {
+      // reCAPTCHA support
+      if (module_exists('recaptcha')) {
+        if ($variables['element']['#type'] == 'captcha' && $variables['element']['#captcha_type'] == 'recaptcha/reCAPTCHA') {
+          if (!empty($variables['element']['captcha_widgets']['captcha_response'])) {
+            $error = ife_errors('get', $variables['element']['captcha_widgets']['captcha_response']['#id']);
+          }
+        }
+      }
+    }
+
     if (!empty($error)) {
       $output .= '<div class="messages error messages-inline">' . $error . '</div>';
     }
