diff --git a/services_views.resource.inc b/services_views.resource.inc
index 3a6aabc..f4c2366 100755
--- a/services_views.resource.inc
+++ b/services_views.resource.inc
@@ -182,7 +182,11 @@ function services_views_execute_view($view_info, $view = NULL, $display_id = NUL
             }
           }
         }
-
+        // Check html_strip property
+        if ($field->options['alter']['strip_tags'] == '1') {
+          // Strip the tags which aren't allowed using strip_tags ( $subject, $preserved_tags ).
+          $obj[$idx] = strip_tags($obj[$idx] , $field->options['alter']['preserve_tags']);
+        }
         // Check cardinality of the field.
         if ($field->field_info['cardinality'] == 1 && count($obj) == 1) {
           $obj = $obj[$idx];
