diff --git a/resources/file_resource.inc b/resources/file_resource.inc
index 28402ca..2dda47e 100644
--- a/resources/file_resource.inc
+++ b/resources/file_resource.inc
@@ -216,7 +216,12 @@ function _file_resource_create_raw() {
     // file_save_upload() validates the file extension and name's length. File
     // size is limited by the upload_max_filesize directive in php.ini.
     $scheme = file_default_scheme();
-    $file = file_save_upload($field_name, array(), "$scheme://");
+    // Set file validators: allowed extension
+    $validators = array();
+    if ($extensions = variable_get('services_allowed_extensions', SERVICES_ALLOWED_EXTENSIONS)) {
+      $validators['file_validate_extensions'] = $extensions;
+    }
+    $file = file_save_upload($field_name, $validators, "$scheme://");
 
     if (!empty($file->fid)) {
       // Change the file status from temporary to permanent.
@@ -433,8 +438,8 @@ function _services_file_check_destination($destination) {
  *   A safe file name and extension.
  */
 function _services_file_check_name_extension($name) {
-  // This whitelist is copied from file_save_upload().
-  $extensions = 'jpg jpeg gif png txt doc xls pdf ppt pps odt ods odp';
+  //Fetch the file extensions set in the variable at the time module is enabled
+  $extensions = variable_get('services_allowed_extensions', SERVICES_ALLOWED_EXTENSIONS);
 
   // Get the part of the name after the last period (".").
   $name = explode('.', $name);
diff --git a/services.module b/services.module
index 0ce7953..e86a316 100644
--- a/services.module
+++ b/services.module
@@ -9,6 +9,7 @@
  * Minimum CTools version needed.
  */
 define('SERVICES_REQUIRED_CTOOLS_API', '1.7');
+define('SERVICES_ALLOWED_EXTENSIONS', 'jpg jpeg gif png txt doc xls pdf ppt pps odt ods odp');
 
 /*
  * Function to return list of batch options
